docs/upgrading_wagtail_v2/index.rst

```diff
@@ -0,0 +1,158 @@
+==============================
+Upgrading to Wagtail Version 2
+==============================
+
+Wagtail 2.0 Requirements
+========================
+* Django versions: 1.11, 2.0 
+* Python versions: 3.4, 3.5, 3.6
+    - It's recommended that you upgrade your python version first to be compatible with python 3.4.
+    
+For more requirements, see `Compatible Django / Python versions`_ table
+
+Upgrade process
+===============
+We recommend upgrading one feature release at a time, even if your project is several versions behind the current one. This has a number of advantages over skipping directly to the newest release:
+
+* If anything breaks as a result of the upgrade, you will know which version caused it, and will be able to troubleshoot accordingly;
+* Deprecation warnings shown in the console output will notify you of any code changes you need to make before upgrading to the following version;
+* Some releases make database schema changes that need to be reflected on your project by running ``./manage.py makemigrations`` - this is liable to fail if too many schema changes happen in one go.
```

Comment:
> this is liable to fail if too many schema changes happen in one go.

I don't think this is true. All migrations are executed in order. So running `manage.py migrate` is the same as running `manage.py migrate app 001 && `manage.py migrate app 002 && `manage.py migrate app 003```. But `manage.py migrate` is better as migration dependencies are respected.

Reply:
This text is copied from the existing page at http://docs.wagtail.io/en/v2.1/releases/upgrading.html. It's referring to things like custom image models, where a change within Wagtail requires a corresponding migration to be created within the user project, using `./manage.py makemigrations`. This relies on Django's migration autodetection logic, which can't handle too many changes at once.

IIRC, this came up around Wagtail 1.8-1.10 when the `filter` field on image renditions was changed from a foreign key to a string - we had to add and populate the new field, switch over the unique constraints, and drop the old field, in three separate migrations over three releases. Skipping straight to the last release would mean you'd end up with one migration containing all of those changes.